### PR TITLE
feat: hide resource variables as paid feature

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio/issues/2532

Screenshot 2024-01-14 at 20 42 44

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/builder/features/settings-panel/variable-popover.tsx [hunk 1]
```diff
@@ -324,6 +329,9 @@ const VariablePanel = forwardRef<
     variable?: DataSource;
   }
 >(({ variable }, ref) => {
+  const userPlanFeatures = useStore($userPlanFeatures);
+  const { allowResourceVariables = false } = userPlanFeatures;
```
Comment (kof) [inline apps/builder/app/builder/features/settings-panel/variable-popover.tsx:333]: default should be in inside userPlanFeatures

File: apps/builder/app/builder/features/settings-panel/variable-popover.tsx [hunk 2]
```diff
@@ -360,17 +368,48 @@ const VariablePanel = forwardRef<
...
-          value === "json" ? "JSON" : humanizeString(value)
-        }
+        options={typeOptions}
+        getLabel={getTypeLabel}
         value={type}
         onChange={setType}
-      />
+      >
+        {typeOptions.map((option) =>
+          option === "resource" && allowResourceVariables === false ? (
```
Comment (kof) [inline apps/builder/app/builder/features/settings-panel/variable-popover.tsx:390]: Nit: I keep single SelectItem and just set disabled and badge conditionally, this notation was hard to read

File: apps/builder/app/shared/db/user-plan-features.server.ts [hunk 3]
```diff
@@ -72,6 +73,7 @@ export const getUserPlanFeatures = async (
 
     return {
       allowShareAdminLinks: true,
+      allowResourceVariables: true,
```
Comment (kof) [inline apps/builder/app/shared/db/user-plan-features.server.ts:76]: This default should be false in my opinion

Comment (kof) [reply apps/builder/app/shared/db/user-plan-features.server.ts:76]: ah sorry, its for pro

File: packages/design-system/src/components/pro-badge.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        borderRadius: theme.borderRadius[2],
+        px: theme.spacing[3],
+        py: theme.spacing[1],
+        height: theme.spacing[9],
+        color: theme.colors.foregroundContrastMain,
+        alignItems: "center",
+        maxWidth: "100%",
+        whiteSpace: "nowrap",
+        overflow: "hidden",
+        // @tood doesn't work in tooltips, needs a workaround
```
Comment (kof) [inline packages/design-system/src/components/pro-badge.tsx:18]:
```suggestion
        // @todo doesn't work in tooltips, needs a workaround
```

File: packages/design-system/src/components/pro-badge.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { theme } from "../stitches.config";
+import { Text } from "./text";
+
+export const ProBadge = () => {
```
Comment (kof) [inline packages/design-system/src/components/pro-badge.tsx:4]: Didn't you want to reuse it without putting it into design system?

**Review by kof — APPROVED** [review]

File: apps/builder/app/builder/features/settings-panel/variable-popover.tsx (hunk 2 above)
Comment (TrySound) [reply apps/builder/app/builder/features/settings-panel/variable-popover.tsx:390]: So you want this condition to be written twice instead?

File: packages/design-system/src/components/pro-badge.tsx (hunk 5 above)
Comment (TrySound) [reply packages/design-system/src/components/pro-badge.tsx:4]: You wanted to put it into design system 🤷

Comment (kof) [reply packages/design-system/src/components/pro-badge.tsx:4]: I was talking about regarding a generic badge/chip component, but idk we do have some not generic component there too, but I guess we don't have to.

**Pull request merged**